### DISCUSSION TITLE
"Lit is in dev mode" message displayed in Production - Angular.

### DISCUSSION THREAD

**Discussion by OvidijusParsiunas** [lit/lit#3671] [Q&A] [answered]
I have a custom element that I developed using the [LitElement TypeScript starter project](https://github.com/lit/lit-element-starter-ts/tree/lit-element-2.x#readme), bundled using the existing `rollup` configuration and published via `npm`.

The weird problem that I am having is that when I use the element in Angular in Production - the "Lit is in dev mode" warning message is still displayed in the console. I am quite confused for if my custom element has been bundled the right way or if I am importing the correct file. What I am more concerned about is that the developers that will use my custom element will also be asking the same thing.

Here is a [Live Example](https://codesandbox.io/s/lit-element-dev-mode-in-prod-gqw2eh?file=/src/main.ts) on the issue where a production environment has been simulated via the `enableProdMode` function and the console still displays the warning.

[Package repo](https://github.com/OvidijusParsiunas/active-table/tree/main/component).

[Package npm](https://www.npmjs.com/package/active-table).

Any help/advise regarding this would be greatly appreciated!

**Comment by augustjk** [comment, marked as answer]
The way that dev mode works for Lit packages is with Node's [conditional exports](https://nodejs.org/api/packages.html#conditional-exports).

I don't think the codesandbox link you provided is a true simulation of a production environment. Calling the `enableProdMode()` function probably only switches the runtime mode for Angular. Lit's dev mode is not runtime based, but rather import based.

I'm not too familiar with how Angular projects are built and served but from what I can find from the docs, running `ng build` should run the builder in production mode  https://angular.io/guide/workspace-config#alternate-build-configurations. It says the default builder uses webpack, which should pick the correct Lit version when run in production mode.

If you need to run the project in production mode with the `ng serve` command, perhaps the `--configuration` option can be set to "production". https://angular.io/cli/serve#options

**Reply by OvidijusParsiunas** [reply]
Hi @augustjk, thankyou for your response! You are absolutely correct, after I pulled the code and built it for prod on my machine - the warning disappeared. Thanks again!
